relayer/crates/starknet-test-components/src/impls/bootstrap_madara/start_pathfinder.rs

```diff
@@ -0,0 +1,92 @@
+use cgp::prelude::*;
+use hermes_cosmos_test_components::bootstrap::traits::chain::start_chain::{
+    ChainFullNodeStarter, ChainFullNodeStarterComponent,
+};
+use hermes_cosmos_test_components::bootstrap::traits::types::chain_node_config::HasChainNodeConfigType;
+use hermes_cosmos_test_components::bootstrap::traits::types::genesis_config::HasChainGenesisConfigType;
+use hermes_runtime_components::traits::fs::create_dir::CanCreateDir;
+use hermes_runtime_components::traits::os::child_process::CanStartChildProcess;
+use hermes_runtime_components::traits::runtime::HasRuntime;
+
+use crate::types::node_config::StarknetNodeConfig;
+
+#[cgp_new_provider(ChainFullNodeStarterComponent)]
+impl<Bootstrap, Runtime> ChainFullNodeStarter<Bootstrap> for StartPathfinder
+where
+    Bootstrap: HasRuntime<Runtime = Runtime>
+        + HasChainNodeConfigType<ChainNodeConfig = StarknetNodeConfig>
+        + HasChainGenesisConfigType
+        + CanRaiseAsyncError<Runtime::Error>,
+    Runtime: CanStartChildProcess + CanCreateDir,
+{
+    async fn start_chain_full_nodes(
+        bootstrap: &Bootstrap,
+        chain_home_dir: &Runtime::FilePath,
+        chain_node_config: &StarknetNodeConfig,
+        chain_genesis_config: &Bootstrap::ChainGenesisConfig,
+    ) -> Result<Vec<Runtime::ChildProcess>, Bootstrap::Error> {
+        let chain_command = Runtime::file_path_from_string("pathfinder");
+
+        let pathfinder_home = Runtime::join_file_path(
+            chain_home_dir,
+            &Runtime::file_path_from_string("pathfinder"),
+        );
+
+        bootstrap
+            .runtime()
+            .create_dir(&pathfinder_home)
+            .await
+            .map_err(Bootstrap::raise_error)?;
+
+        let gateway_port = chain_node_config.rpc_port + 1;
+
+        // Use RPC Port + 2 for Anvil port for now
+        let anvil_port = chain_node_config.rpc_port + 2;
+
+        // Use RPC Port + 3 for Pathfinder port for now
+        let pathfinder_port = chain_node_config.rpc_port + 3;
+
+        let args = [
+            "--data-directory",
+            &Runtime::file_path_to_string(&pathfinder_home),
+            "--network",
+            "custom",
+            "--storage.state-tries",
+            "archive",
+            "--ethereum.url",
+            &format!("http://localhost:{anvil_port}"),
+            "--gateway-url",
+            &format!("http://localhost:{gateway_port}/gateway"),
+            "--feeder-gateway-url",
+            &format!("http://localhost:{gateway_port}/feeder_gateway"),
+            "--chain-id",
+            "starknet-devnet",
```

Comment:
.. but here, it is `starknet-devnet`.

Reply:
Yeah I'm not quite sure why we can set the chain ID here. Ideally it should be customizable on all 3 places, but right now it is only customizable here. It also don't appear to have any effect, since it is acting as a full node.